modules/components/pages/selfservice.adoc

```diff
@@ -0,0 +1,706 @@
+
+* `name`: `"userDetails"`
+* `identityEmailField` - field name for the identity email address
+
+.Example configuration
+[source,json]
+----
+{
+  "name" : "userDetails",
+  "identityEmailField" : "mail"
+}
+----
+
+=== User Query Stage
+
+Stage is responsible for querying the configured identity service for a user based on the provided query fields.
+Once identified, it populates `mail` and `userId` fields in process context.
+
+Configuration:
+
+* `name`: `"userQuery"`
+* `validQueryFields` - list of query fields to be used when looking up the user
+* `identityServiceUrl` - identity service URL used to lookup the user
+* `identityIdField` - field name for the identity ID
+* `identityEmailField` - field name for the identity email address
+* `identityUsernameField` - field name for the identity username
+* `identityAccountStatusField` - field name for the identity account status
+
+.Example configuration
+[source,json]
+----
+{
+  "name" : "userQuery",
+  "validQueryFields" : ["userName", "mail"],
+  "identityServiceUrl" : "managed/user",
+  "identityIdField" : "_id",
+  "identityEmailField" : "mail",
+  "identityUsernameField" : "userName",
+  "identityAccountStatusField" : "accountStatus"
+}
+----
+
+=== Validate Active Account Stage
+
+Stage responsible for validating user account status before password reset.
+
+Configuration:
+
+* `name`: `"validateActiveAccount"`
+* `validStatusValue` - account status value that is considered valid
+// Following property is never used by the stage
+// * `accountStatusField` - field name for the identity status
+
+.Example configuration
+[source,json]
+----
+{
+  "name" : "validateActiveAccount",
+  "validStatusValue" : "active"
+}
+----
+
+=== Social User Details Stage
+
+Stage responsible for gathering social user profile details.
+It expects the `mail` field to be populated in process context which it uses to verify against the email address specified in the provided in user object.
+
+Configuration:
+
+* `name`: `"socialUserDetails"`
+* `identityEmailField` - field name for the identity email address
+* `providers` - list of identity provider configurations
+** `name` - unique provider name
+** `type` - authentication type (e.g., `OPENID_CONNECT`, `OAUTH`)
+** `icon` - icon HTML
+** `authorization_endpoint` - endpoint for authentication and authorization of a user
+** `token_endpoint` - endpoint for requesting access and ID tokens
+** `userinfo_endpoint` - endpoint for requesting user information
+** `well-known` - well-known endpoint for OpenID Connect configuration key-value pairs
+** `client_id` - OAuth client ID
+** `client_secret` - OAuth client secret
+** `scope` - OAuth scopes being requested
+*** expected value: `List<scope>`
+** `authenticationId` - property that maps to unique user identifier
+// Following property is never used by the stage
+// ** `schema` - JSON Schema for generating form fields
+** `propertyMap` - property mapping from provider fields to OpenIDM fields (optional)
```

Comment:
This is copied from a source code comment, I'm wondering if it shouldn't mentione "Wren:IDM fields" instead of "OpenIDM".

Reply:
Yes please.